addons/mail/models/discuss/discuss_channel.py

```diff
@@ -505,10 +508,12 @@ def add_members(
                     payload["invited_by_user_id"] = self.env.user.id
                 member._bus_send("discuss.channel/joined", payload)
                 if post_joined_message:
+                    message_author = current_partner if current_partner else current_guest
+                    member_name, author_name = self.get_displayed_name(member=member, author=message_author)
                     notification = (
-                        _("joined the channel")
-                        if member.is_self
-                        else _("invited %s to the channel", member._get_html_link(for_persona=True))
+                        _("%(user)s joined the thread", user=member_name) if member.is_self and self.parent_channel_id else
+                        _("%(user)s joined the channel", user=member_name) if member.is_self else
+                        _("%(user1)s invited %(user2)s to the %(target)s", user1=author_name, user2=member._get_html_link(for_persona=True), target="thread" if self.parent_channel_id else "channel")
```

Comment:
Why do we make user with a link on some case but not others? It feels we should be consistent.